pkg/network/payload/notary_request.go

```diff
@@ -136,3 +136,12 @@ func (r *P2PNotaryRequest) isValid() error {
 	}
 	return nil
 }
+
+// Copy creates a deep copy of the P2PNotaryRequest.
+func (r *P2PNotaryRequest) Copy() *P2PNotaryRequest {
```

Comment:
A couple of words about cached values are needed in the doc, like for transaction's `Copy()`.